plugins/de.cau.cs.kieler.klighd.standalone/pom.xml

```diff
@@ -16,7 +16,7 @@
   <dependencies>
     <dependency>
       <groupId>org.eclipse.elk</groupId>
-      <artifactId>org.eclipse.elk.core.service</artifactId>
+      <artifactId>org.eclipse.elk.core</artifactId>
```

Comment:
Why do you remove elk.core.service here? Is it not needed?

Reply:
Looking at the dependencies in the manifest, this dependency was wrong in the first place (the .service part is not required there, but the elk.core bundle directly). Furthermore, the only class in the package has only dependencies to elk.core and not elk.core.service, so this is correct like this.
Looking at the last build with this [here](http://services.typefox.io/open-source/jenkins/job/KLighD/job/master/48/) also shows some issue in the maven build with this dependency, which is resolved with this change.

Reply:
This change is wrong, as the class `LightDiagramServices` requires `DiagramLayoutEngine` from `elk.service`, see https://github.com/kieler/KLighD/blob/cc19356b0d50d76390d4444ca186c095572b7637/plugins/de.cau.cs.kieler.klighd/src/de/cau/cs/kieler/klighd/LightDiagramServices.java#L246-L261

For compiling `elk.service` is include via a Manifest dependency.
For this reason, `elk.service` is been published to maven central, see https://github.com/eclipse/elk/pull/511
This, please re-add the dependency to `elk.service`, otherwise Klighd won't work in standalone mode "out-of-the-box".

Reply:
Wrt.
> Furthermore, the only class in the package has only dependencies to elk.core and not elk.core.service, so this is correct like this.

This project with its dependencies serves as a so-called Bill-Of-Materials (BOM), see https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html

The aim is that only a dependency to this project is required in order to bring Klighd to work.

Reply:
Okay, I did not know abot the BOM behavior of the project, so this makes sense as it was before. The initial reason why I changed this though is because the build suddenly broke as mentioned before and seen [here](http://services.typefox.io/open-source/jenkins/job/KLighD/job/master/48/) and this at least fixed the Maven build again.
The issue in the build is summarized in the logs:
```
No versions are present in the repository for the artifact with a range [4.5.1,6.0.0)
  com.sun.jna:com.sun.jna:jar:null

from the specified remote repositories:
  klighd (http://kieler.github.io/KLighD/repo, releases=true, snapshots=true),
  central (https://repo.maven.apache.org/maven2, releases=true, snapshots=false)
Path to dependency: 
	1) de.cau.cs.kieler.klighd:de.cau.cs.kieler.klighd.standalone:pom:2.0.0.v20200918
	2) org.eclipse.elk:org.eclipse.elk.core.service:jar:0.7.0
	3) org.eclipse.platform:org.eclipse.ui.workbench:jar:3.110.0
	4) org.eclipse.platform:org.eclipse.e4.ui.workbench.addons.swt:jar:1.3.1100
	5) org.eclipse.platform:org.eclipse.e4.ui.workbench.renderers.swt:jar:0.14.1300
	6) org.eclipse.platform:org.eclipse.e4.ui.workbench.swt:jar:0.14.1100
	7) org.eclipse.platform:org.eclipse.urischeme:jar:1.1.100

```

As I now tracked this down, the Issue is only has nothing to do with the merge of keith and would have occured otherwise as well because of the [org.eclipse.urischeme](https://mvnrepository.com/artifact/org.eclipse.platform/org.eclipse.urischeme) transitive dependency of elk.core.service, which had a new release 1.1.100 on Sept. 16th and introduced a new dependency to com.sun.jna version [4.5.1,6.0.0), which is not available on Maven Central. Only a previous version 3.0.5 is available there with a note to use net.java.dev.jna instead (see [here](https://mvnrepository.com/artifact/com.sun.jna/com.sun.jna)).
As this new dependency is automatically resolved by maven, the build currently fails when re-adding the dependency to elk.core.service again.

I don't quite know enough about maven's own dependency management, as we otherwise only use the depenendy mechanism of tycho, so have not found a solution to this problem yet. Do you know how to resolve this now-broken transitive dependency? Sadly, this seems to be blocking the next KLighD release.

Reply:
As this problem also affects the ELK nightly build, I submitted an Issue on the Eclipse bug tracker:
[https://bugs.eclipse.org/bugs/show_bug.cgi?id=567244](https://bugs.eclipse.org/bugs/show_bug.cgi?id=567244)

Reply:
> Do you know how to resolve this now-broken transitive dependency? Sadly, this seems to be blocking the next KLighD release.

Can we have a call on that today (Sept 23rd) afternoon?